### PR TITLE
fix(files_sharing): Parse OCM share permissions from OCM and not OCS prop

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

Checks for OCM permissions prop, but parses them from the OCS permissions prop.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_sharing/lib/External/Storage.php [hunk 1]
```diff
@@ -345,14 +345,17 @@ public function isSharable($path): bool {
 
 	public function getPermissions($path): int {
 		$response = $this->propfind($path);
+		$ocsPermissions = $response['{http://open-collaboration-services.org/ns}share-permissions'];
```
Comment (come-nc) [inline apps/files_sharing/lib/External/Storage.php:348]:
```suggestion
		$ocsPermissions = $response['{http://open-collaboration-services.org/ns}share-permissions'] ?? null;
```
Is something like that not needed? What’s the type of `$response`?

**Review by come-nc — COMMENTED** [review]
Ok for the logic, but error handling is missing, $reponse could be false, or could be missing the keys.

File: apps/files_sharing/lib/External/Storage.php (hunk 1 above)
Comment (provokateurin) [reply apps/files_sharing/lib/External/Storage.php:348]: Right, missing key already results in null but also logs an error message.

**Review by come-nc — APPROVED** [review]

**Review by kesselb — APPROVED** [review]

**Pull request merged**